gobblin-service/src/main/java/org/apache/gobblin/service/modules/restli/GobblinServiceFlowExecutionResourceHandlerWithWarmStandby.java

```diff
@@ -40,16 +40,18 @@
 import org.apache.gobblin.service.FlowStatusId;
 import org.apache.gobblin.service.modules.core.GobblinServiceManager;
 import org.apache.gobblin.service.modules.orchestration.DagActionStore;
+import org.apache.gobblin.service.modules.orchestration.DagManagementStateStore;
+
 
 @Slf4j
 public class GobblinServiceFlowExecutionResourceHandlerWithWarmStandby extends GobblinServiceFlowExecutionResourceHandler{
-  private DagActionStore dagActionStore;
+  private DagManagementStateStore dagManagementStateStore;
   @Inject
   public GobblinServiceFlowExecutionResourceHandlerWithWarmStandby(FlowExecutionResourceLocalHandler handler,
       @Named(GobblinServiceManager.SERVICE_EVENT_BUS_NAME) EventBus eventBus,
-      Optional<HelixManager> manager, @Named(InjectionNames.FORCE_LEADER) boolean forceLeader, DagActionStore dagActionStore) {
+      Optional<HelixManager> manager, @Named(InjectionNames.FORCE_LEADER) boolean forceLeader, DagManagementStateStore dagManagementStateStore) {
     super(handler, eventBus, manager, forceLeader);
```

Comment:
do we want to always encapsulate the dagAction additional/removal/checking through dagManagement? it seems like it from the PR that no reference to it should remain outside of dagManagement, however dagManagement provides a lot more functionality as well I'm not sure we want to give access to everywhere. what do you think?

Reply:
Hmm, I do not have a strong opinion about anything.

Reply:
Okay after we discussed today this makes sense to me